process-migration-service/src/test/java/org/kie/processmigration/service/util/ConfigurationValueProducer.java

```diff
@@ -0,0 +1,28 @@
+package org.kie.processmigration.service.util;
+
+import java.io.FileNotFoundException;
+import java.util.Map;
+
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.InjectionPoint;
+
+import org.wildfly.swarm.spi.runtime.annotations.ConfigurationValue;
+import org.yaml.snakeyaml.Yaml;
+
+public class ConfigurationValueProducer {
+
+    Map<String, String> configuration;
+
+    @SuppressWarnings("unchecked")
+    public ConfigurationValueProducer() throws FileNotFoundException {
+        Yaml yaml = new Yaml();
+        configuration = yaml.loadAs(this.getClass().getClassLoader().getResourceAsStream("project-defaults.yml"), Map.class);
```

Comment:
When I tried to run this and print out the map, I found out that it is actually a LinkedHashMap with values of LinkedHashMap etc. so values cannot be typed to String in runtime. This should be probably done differently.

Reply:
Yeah, the tests are totally screwed up. I will come back to this as soon as possible

Reply:
> Regarding the authentication, the way I understand it is that we just store credentials on the migration service side and then we will forward them to KIE Server. So plans, migration and migrationdefinition can be created without any authentication? And currently it can connect to just one KIE Server, right?

Yes, I don't consider authentication is required at the Migration service level to avoid re-implementing all the auth functionalities already in place for the workbench and Kie server until a common SSO functionality is provided. WDYT?